pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -446,11 +446,8 @@ func (p *ConfigurationEditFlags) Apply(
 		if cmd.Flags().Changed("annotation") && containsAnnotation(p.Annotations, autoscaling.InitialScaleAnnotationKey) {
 			return fmt.Errorf("only one of the --scale-init or --annotation %s can be specified", autoscaling.InitialScaleAnnotationKey)
 		}
-		annotationsMap := map[string]string{
-			autoscaling.InitialScaleAnnotationKey: strconv.Itoa(p.ScaleInit),
-		}
 
-		err = servinglib.UpdateAnnotations(service, template, annotationsMap, []string{})
+		err = servinglib.UpdateRevisionTemplateAnnotation(template, autoscaling.InitialScaleAnnotationKey, strconv.Itoa(p.ScaleInit))
```

Comment:
IMO, we should have two utils
 - `UpdateRevisionAnnotations`
 - `UpdateServiceAnnotations`
Each taking respective object, toUpdate and toRemove map.
The callers of these utils can then decide if the given annotations should be set on either or both (explaining why in comments).

Reply:
In the current codebase there's only one place that `UpdateAnnotations` is used and needs to be changed to filter autoscaling annotations. I tried to aim for quick win to unblock CI etc. However, I do agree that separate `Update*` util methods are likely better solution to expose filtering logic at caller level. I'll update the PR shortly. :wink: